src/media/js/elements/select.js

```diff
@@ -0,0 +1,218 @@
+                mktOption.removeAttribute('data-mkt-option--index');
+            } else {
+                mktOption.setAttribute('data-mkt-option--index', index++);
+            }
+        }
+    }
+
+    function copyAttrs(dest, source) {
+        // Copies attributes from one element to another.
+        for (var i = 0; i < source.attributes.length; i++) {
+            var attr = source.attributes[i];
+            if (attr !== 'id') {
+                dest.setAttribute(attr.name, attr.value);
+            }
+        }
+        return dest;
+    }
+
+    function proxyElement(htmlElement, customElement, key) {
+        // Proxy custom element interface to actual element.
+        Object.getOwnPropertyNames(htmlElement.prototype).forEach(function(method) {
+            try {
+                customElement.prototype[method] = {
+                    value: function() {
+                        return this[key][method]();
+                    }
+                };
+            } catch (e) {
+                Object.defineProperty(customElement.prototype, method, {
+                    get: function() {
+                        return this[key][method];
+                    }
+                });
+            }
+        });
+    }
+
+    proxyElement(HTMLSelectElement, MktSelectElement, 'select');
+    proxyElement(HTMLOptGroupElement, MktOptGroupElement, 'optGroup');
+    proxyElement(HTMLOptionElement, MktOptionElement, 'option');
+
+    z.body.on('click', 'mkt-selected', function() {
+        this.parentNode.toggle();
+    })
+
+    .on('change', 'mkt-select', function() {
+        // Update <mkt-selected-text>.
+        var newText = this.querySelector('mkt-option[selected]').innerHTML;
+        this.parentNode.querySelector('mkt-selected-text').innerHTML = newText;
+    })
+
+    .on('click', 'mkt-option', function() {
+        var mktSelect = this.parentNode.parentNode;
+        mktSelect.toggle();
+
+        // Change value of hidden select element.
+        var select = mktSelect.select;
+        select.value = this.value;
+
+        // Change [selected] of <option>s.
+        var options = select.querySelectorAll('option');
+        for (var i = 0; i < options.length; i++) {
+            options[i].removeAttribute('selected');
+        }
+        this.option.setAttribute('selected', '');
+
+        // Change [selected] of <mkt-option>s. Update [data-index]es.
+        updateMktOptionIndices(mktSelect, this);
+        var mktOptions = mktSelect.querySelectorAll('mkt-option');
+        for (i = 0; i < mktOptions.length; i++) {
+            mktOptions[i].removeAttribute('selected');
+        }
+        this.setAttribute('selected', '');
+
+        $(mktSelect).trigger('change');
+    });
+
+    z.win.on('resize', _.debounce(function() {
+        // Re-align options on resize.
+        var mktSelects = document.querySelectorAll('mkt-select');
+        for (var i = 0; i < mktSelects.length; i++ ) {
+            alignMktOptions(mktSelects[i]);
```

Comment:
It would be nice to not have to do this. I can't think of a way that would definitely work though.